reconcile/openshift_users.py

```diff
@@ -1,35 +1,63 @@
 import itertools
 import logging
+import sys
+from collections.abc import (
+    Callable,
+    Iterable,
+    Mapping,
+)
+from typing import (
+    Any,
+    Optional,
+)
 
 from sretoolbox.utils import threaded
 
 from reconcile import (
     openshift_groups,
     openshift_rolebindings,
-    queries,
 )
+from reconcile.gql_definitions.common.clusters_minimal import (
+    ClusterAuthOIDCV1,
+    ClusterV1,
+)
+from reconcile.status import ExitCodes
+from reconcile.typed_queries.app_interface_vault_settings import (
+    get_app_interface_vault_settings,
+)
+from reconcile.typed_queries.clusters_minimal import get_clusters_minimal
 from reconcile.utils.defer import defer
-from reconcile.utils.oc import OC_Map
+from reconcile.utils.oc_map import (
+    OCLogMsg,
+    OCMap,
+    init_oc_map_from_clusters,
+)
+from reconcile.utils.secret_reader import create_secret_reader
 
 QONTRACT_INTEGRATION = "openshift-users"
 
 
-def get_cluster_users(cluster, oc_map, clusters):
+def get_cluster_users(
+    cluster: str, oc_map: OCMap, clusters: Iterable[ClusterV1]
+) -> list[dict[str, Any]]:
     oc = oc_map.get(cluster)
-    if not oc:
+    if isinstance(oc, OCLogMsg):
         logging.log(level=oc.log_level, msg=oc.message)
         return []
+    if not oc:
+        logging.error("No OC client for cluster %s", cluster)
+        return []
     users: list[str] = []
 
     # get cluster info for current cluster name from clusters list
-    cluster_info = next((cl for cl in clusters if cl["name"] == cluster))
+    cluster_info = next((cl for cl in clusters if cl.name == cluster))
 
     # backwarts compatibiltiy for clusters w/o auth
     identity_prefixes = ["github"]
 
-    for auth in cluster_info["auth"]:
-        if auth["service"] == "oidc":
-            identity_prefixes.append(auth["name"])
+    for auth in cluster_info.auth:
+        if auth.service == "oidc" and isinstance(auth, ClusterAuthOIDCV1):
```

Comment:
Please do just the `isinstance` check, it should enough here.

Reply:
Yeah was actually wondering about this one. Removed it now - thanks you! :)